tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,318 @@
+    data = {"x": [], "y": [2, 3]}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_lists():
+    """Test expected output from one level nested lists in a dictionary's values"""
+    data = {"x": [[2, 3], [4, 3]]}
+    expected = pd.DataFrame([[2, 3], [4, 3]]).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_lists_all_scalar():
+    """Test that all values in a list in dictionary's values are scalar"""
+    data = {"x": [2, 3, 4, 5, "ragnar"]}
+    expected = {"x": [2, 3, 4, 5, "ragnar"]}
+    assert _check_instance(data)[-1] == expected
+
+
+def test_lists_not_all_scalar():
+    """Trigger error if values in a list in the dictionary's values are not scalar"""
+    data = {"x": [[2, 3], 4, 5, "ragnar"]}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_computation_output_1():
+    """Test output if entry contains no dataframes/series"""
+    data = {"x": range(1, 4), "y": [1, 2]}
+    expected = pd.DataFrame({"x": [1, 1, 2, 2, 3, 3], "y": [1, 2, 1, 2, 1, 2]})
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_computation_output_2():
+    """Test output if entry contains only dataframes/series"""
+    data = {
+        "df": pd.DataFrame({
+            "x": range(1, 6),
+            "y": [5, 4, 3, 2, 1]
+        }),
+        "df1": pd.DataFrame({
+            "x": range(4, 7),
+            "y": [6, 5, 4]
+        }),
+    }
+
+    expected = pd.DataFrame({
+        "df_x": [1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5],
+        "df_y": [5, 5, 5, 4, 4, 4, 3, 3, 3, 2, 2, 2, 1, 1, 1],
+        "df1_x": [4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6, 4, 5, 6],
+        "df1_y": [6, 5, 4, 6, 5, 4, 6, 5, 4, 6, 5, 4, 6, 5, 4],
+    })
+
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_computation_output_3():
+    """Test mix of dataframes and lists"""
+    data = {
+        "df": pd.DataFrame({
+            "x": range(1, 3),
+            "y": [2, 1]
+        }),
+        "z": range(1, 4),
+    }
+    expected = pd.DataFrame({
+        "df_x": [1, 1, 1, 2, 2, 2],
+        "df_y": [2, 2, 2, 1, 1, 1],
+        "z": [1, 2, 3, 1, 2, 3],
+    })
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_computation_output_4():
+    """ Test output from list of strings"""
+    data = {"l1": list(ascii_lowercase[:3]), "l2": list(ascii_uppercase[:3])}
+    expected = pd.DataFrame({
+        "l1": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+        "l2": ["A", "B", "C", "A", "B", "C", "A", "B", "C"],
+    })
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_df_key():
+    """ Raise error if dataframe key is not supplied"""
+    df = pd.DataFrame({"x": [2, 3]})
+    others = {"df": pd.DataFrame({"x": range(1, 6), "y": [5, 4, 3, 2, 1]})}
+
+    with pytest.raises(KeyError):
+        assert expand_grid(df, others=others)
```

Comment:
Assertion not necessary.

```suggestion
        expand_grid(df, others=others)
```